assembl/static2/js/app/components/cookies/acceptCookiesModal.jsx

```diff
@@ -0,0 +1,190 @@
+      modalIsChecked: !prevState.modalIsChecked
+    }));
+  };
+
+  getLegalContentsToAccept = (): LegalContentsArray => {
+    const { hasTermsAndConditions, hasPrivacyPolicy, hasUserGuidelines } = this.props;
+    const legalContentsToAccept = {
+      ACCEPT_CGU: hasTermsAndConditions,
+      ACCEPT_PRIVACY_POLICY_ON_DISCUSSION: hasPrivacyPolicy,
+      ACCEPT_USER_GUIDELINE_ON_DISCUSSION: hasUserGuidelines
+    };
+    const filteredLegalContentsToAccept = LEGAL_CONTENTS_TO_ACCEPT.filter(contentType => legalContentsToAccept[contentType]);
+    return filteredLegalContentsToAccept;
+  };
+
+  acceptAllLegalContents = () => {
+    const legalContentsToAccept = this.getLegalContentsToAccept();
+    this.props.updateAcceptedCookies({
+      variables: { actions: legalContentsToAccept }
+    });
+  };
+
+  closeModal = () => {
+    this.setState({ showModal: false });
+  };
+
+  render() {
+    const { hasTermsAndConditions, hasPrivacyPolicy, hasUserGuidelines } = this.props;
+    const { showModal, modalIsChecked } = this.state;
+    const slug = getDiscussionSlug();
+    const legalContentsToAcceptByRouteName = {
+      terms: hasTermsAndConditions,
+      privacyPolicy: hasPrivacyPolicy,
+      userGuidelines: hasUserGuidelines
+    };
+    const legalContentsArray = Object.keys(legalContentsToAcceptByRouteName).map(
+      key => (legalContentsToAcceptByRouteName[key] ? key : null)
+    );
+    // This array gathers all the legal contents to accept by their route name
+    const cleanLegalContentsArray = legalContentsArray.filter(el => el !== null);
+
+    // TODO: deal with the backcross 'static' logic with this new Modal
+
+    return cleanLegalContentsArray ? (
+      <Modal show={showModal}>
+        <Modal.Header>
+          <Modal.Title>
+            <Translate value="legalContentsModal.title" />
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <FormGroup className="justify">
+            <Checkbox onChange={this.handleModalCheckbox} checked={modalIsChecked} type="checkbox" inline>
+              <Translate value="legalContentsModal.iAccept" /> {/* $FlowFixMe */}
+              <LegalContentsLinksList legalContentsList={cleanLegalContentsArray} />
+            </Checkbox>
+          </FormGroup>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            key="cancel"
+            onClick={this.closeModal}
+            className="button-cancel button-dark"
+            href={`${getContextual('oldLogout', { slug: slug })}?next=${get('home', { slug: slug })}`}
+          >
+            <Translate value="refuse" />
+          </Button>
+          <Button
+            disabled={!modalIsChecked}
+            key="accept"
+            className="button-submit button-dark"
+            onClick={() => {
+              try {
+                this.acceptAllLegalContents();
+              } catch (error) {
+                console.error(error); // eslint-disable-line
```

Comment:
Why do you need this try/catch? If this component is "below" the `GlobalErrorBoundary`, an error that is raised would be handled by this component and sent to sentry

Reply:
Oh yeah good point, that's convenient :)